src/pages/admin/student/edit/components/editStudentComponent.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { Student } from '../../../../../model/student';
+
+interface IProps {
+  student: Student;
+  updateStudent: (property: string, value: any) => void;
+  saveStudent: (event: any) => void;
+}
+
+function updateStudentFullName(e: any) {
+  this.props.updateStudent('fullname', e.target.value);
+}
+
+function updateStudentEmail(e: any) {
+  this.props.updateStudent('email', e.target.value);
+}
+
+function updateStudentIsActive(e: any) {
+  this.props.updateStudent('isActive', e.target.value);
+}
+
+function saveStudent(e: any) {
+  this.props.saveStudent(event);
+}
+
+export class EditStudentComponent extends React.Component<IProps, {}>  {
+  public render() {
+    return (
+      <div>
+        <input
+          type="text"
+          value={this.props.student.fullname}
+          onChange={updateStudentFullName.bind(this)}
```

Comment:
Performance tip, you have to move this bind to the constrcutor, if we don't move this to the constructor it will build the bind on each render. @crsanti can provide you with a sample